ringbuf/reader.go

```diff
@@ -0,0 +1,274 @@
+package ringbuf
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"runtime"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+var (
+	errClosed = errors.New("perf reader was closed")
+	errEOR    = errors.New("end of ring")
+	errBusy   = errors.New("sample not committed yet")
+)
+
+func addToEpoll(epollfd, fd int) error {
+
+	event := unix.EpollEvent{
+		Events: unix.EPOLLIN,
+		Fd:     int32(fd),
+	}
+
+	if err := unix.EpollCtl(epollfd, unix.EPOLL_CTL_ADD, fd, &event); err != nil {
+		return fmt.Errorf("can't add fd to epoll: %v", err)
+	}
+	return nil
+}
+
+// ringbufHeader from 'struct bpf_ringbuf_hdr' in kernel/bpf/ringbuf.c
+type ringbufHeader struct {
+	Len   uint32
+	PgOff uint32
```

Comment:
This doesn't seem to be used by the code, why?

Reply:
it's used in `readRecord`